engines/wage/script.h

```diff
@@ -0,0 +1,178 @@
+
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ * MIT License:
+ *
+ * Copyright (c) 2009 Alexei Svitkine, Eugene Sandulenko
+ *
+ * Permission is hereby granted, free of charge, to any person
+ * obtaining a copy of this software and associated documentation
+ * files (the "Software"), to deal in the Software without
+ * restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following
+ * conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ * OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ * WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ *
+ */
+
+#ifndef WAGE_SCRIPT_H
+#define WAGE_SCRIPT_H
+
+namespace Wage {
+
+class Script {
+public:
+	Script(Common::SeekableReadStream *data);
+	~Script();
+
+private:
+	Common::SeekableReadStream *_data;
+
+	WageEngine *_engine;
+	World *_world;
+	int _loopCount;
+	Common::String *_inputText;
+	Designed *_inputClick;
+	bool _handled;
+
+	class Operand {
+	public:
+		union {
+			Obj *obj;
+			Chr *chr;
+			Designed *designed;
+			Scene *scene;
+			int16 number;
+			Common::String *string;
+			Designed *inputClick;
+		} _value;
+		OperandType _type;
+		Common::String _str;
+
+		Operand(Obj *value, OperandType type) {
+			_value.obj = value;
+			_type = type;
+		}
+
+		Operand(Chr *value, OperandType type) {
+			_value.chr = value;
+			_type = type;
+		}
+
+		Operand(Scene *value, OperandType type) {
+			_value.scene = value;
+			_type = type;
+		}
+
+		Operand(int value, OperandType type) {
+			_value.number = value;
+			_type = type;
+		}
+
+		Operand(Common::String *value, OperandType type) {
+			_value.string = value;
+			_type = type;
+		}
+
+		Operand(Designed *value, OperandType type) {
+			_value.inputClick = value;
+			_type = type;
+		}
+
+		~Operand() {
+			if (_type == STRING)
+				delete _value.string;
+		}
+
+		Common::String toString() {
+			char buf[128];
+
+			if (_value.obj == NULL)
```

Comment:
This check looks odd, what is the purpose? It looks like none of the cases below would check for a `NULL` pointer and thus crash anyway?